lib/puppetserver/ca/config/puppet.rb

```diff
@@ -77,8 +77,9 @@ def load(cli_overrides: {}, logger:, ca_dir_warn: true)
         end
 
         def default_certname
-          hostname = Facter.value(:hostname)
-          domain = Facter.value(:domain)
+          networking = Facter.value(:networking)
+          hostname = networking[:hostname]
+          domain = networking[:domain]
           if domain and domain != ''
             fqdn = [hostname, domain].join('.')
```

Comment:
wouldn't it be easier to always use the networking.fqdn fact? I think when there's no domain, facter will set the fqdn to the hostname?

Reply:
That seems reasonable.

Reply:
Im assuming this was copied from puppet https://github.com/puppetlabs/puppet/blob/7739378f6fa840d6dbee237132754c64f94cc38a/lib/puppet/settings.rb#L72